RM-949 PMM 2.21.0 Release Notes (#574)
percona/pmm-doc: docs/release-notes/2.21.0.md

# Percona Monitoring and Management 2.21.0

<table class="docutils field-list" frame="void" rules="none">
  <colgroup>
    <col class="field-name">
    <col class="field-body">
  </colgroup>
  <tbody valign="top">
    <tr class="field-odd field">
      <th class="field-name">Date:</th>
      <td class="field-body">August 24, 2021</td>
    </tr>
    <tr class="field-even field">
      <th class="field-name">Installation:</th>
      <td class="field-body">
        <a class="reference external" href="https://www.percona.com/software/pmm/quickstart">Installing Percona Monitoring and Management</a></td>
    </tr>
  </tbody>
</table>

Percona Monitoring and Management (PMM) is a free and open-source platform for managing and monitoring MySQL, MongoDB, and PostgreSQL performance.

## New Features

- [PMM-8158](https://jira.percona.com/browse/PMM-8158): Backup Management: Delete artifact from storage — When deleting a backup from Backup Management, a new option lets you also delete the data file from storage.
- [PMM-8156](https://jira.percona.com/browse/PMM-8156): Backup Management: Retention — You can now define how many of the most recent scheduled backups to keep.

> - [PMM-8214](https://jira.percona.com/browse/PMM-8214): Ability to collect Kubernetes cluster metrics — Collection only–metrics are not currently presented on any PMM dashboard.

Reviewer: `only–metrics` this looks weird, I have corrected additional text in Jira to `only, metrics`

Author: Thanks, it's better now, seen in context.

# Percona Monitoring and Management 2.21.0

<table class="docutils field-list" frame="void" rules="none">
  <colgroup>
    <col class="field-name">
    <col class="field-body">
  </colgroup>
  <tbody valign="top">
    <tr class="field-odd field">
      <th class="field-name">Date:</th>
      <td class="field-body">August 24, 2021</td>
    </tr>
    <tr class="field-even field">
      <th class="field-name">Installation:</th>
      <td class="field-body">
        <a class="reference external" href="https://www.percona.com/software/pmm/quickstart">Installing Percona Monitoring and Management</a></td>
    </tr>
  </tbody>
</table>

Percona Monitoring and Management (PMM) is a free and open-source platform for managing and monitoring MySQL, MongoDB, and PostgreSQL performance.

## New Features

> - [PMM-8158](https://jira.percona.com/browse/PMM-8158): Backup Management: Delete artifact from storage — When deleting a backup from Backup Management, a new option lets you also delete the data file from storage.

Author: It's preferred to keep RNs and Jira in sync. Doing that for this suggestion makes the Jira summary long. The original splits on the M-dash with summary as first part, 'additional' text as second. So another suggestion might be: 

```suggestion
- [PMM-8158](https://jira.percona.com/browse/PMM-8158): Backup Management: Delete option — When deleting a backup from Backup Management a new option lets you also delete the data file from storage.
```

- [PMM-8156](https://jira.percona.com/browse/PMM-8156): Backup Management: Retention — You can now define how many of the most recent scheduled backups to keep.

> - [PMM-8214](https://jira.percona.com/browse/PMM-8214): Ability to collect Kubernetes cluster metrics — Collection only, metrics are not currently presented on any PMM dashboard.

Author: @BupycHuk ?

- [PMM-7477](https://jira.percona.com/browse/PMM-7477): Support custom TLS certificates when monitoring remote MongoDB instances

> - [PMM-7888](https://jira.percona.com/browse/PMM-7888): Custom TLS certificates allow SSL connections to monitored PostgreSQL instances (e.g. on Heroku) (Thanks to Jyoti Prakash for reporting this issue)

Author: Heroku example because that is bug reporters actual use case. I thought it a good idea to retain some related keyword for benefit of originator.

## Improvements

- [PMM-8267](https://jira.percona.com/browse/PMM-8267): Backup Management: Active progress indicator
- [PMM-8549](https://jira.percona.com/browse/PMM-8549): Backup Management: Show loading status on delete window
- [PMM-8542](https://jira.percona.com/browse/PMM-8542): Backup Management: Inform that times should be entered in UTC timezone format

> - [PMM-8316](https://jira.percona.com/browse/PMM-8316): DBaaS: PSMDB 1.9 operator support

Author: I will add as 'additional' but, with your permission, without the exclamation mark. But! If you feel we need a bit more pazzaz (as per https://perconacorp.slack.com/archives/G2HEW592P/p1628742239100200) please say.

> - [PMM-8316](https://jira.percona.com/browse/PMM-8316): DBaaS: PSMDB 1.9 operator support

Author: ```suggestion
- [PMM-8316](https://jira.percona.com/browse/PMM-8316): DBaaS: PSMDB 1.9 operator support—For what's new see [release notes](https://www.percona.com/doc/kubernetes-operator-for-psmongodb/RN/Kubernetes-Operator-for-PSMONGODB-RN1.9.0.html).
```

> - [PMM-8386](https://jira.percona.com/browse/PMM-8386): Overview text on Home page omits mentioning PostgreSQL is supported

Author: ```suggestion
- [PMM-8386](https://jira.percona.com/browse/PMM-8386): Overview text on Home page adds mention that PostgreSQL is supported
```

> - [PMM-8386](https://jira.percona.com/browse/PMM-8386): Overview text on Home page omits mentioning PostgreSQL is supported

Author: Improvement not bug so right to refocus. 'Call out' is used in the sense of 'shout out to all our homies' but could be confused with the name of a UI element. So I suggest just to humbly *mention* it.

> - [PMM-7612](https://jira.percona.com/browse/PMM-7612): Integrated Alerting: Validate 'From' email address format (Settings → Communication → Email) 

Author: Point here is the validation. Where it happens is an FYI aside.

> - [PMM-7612](https://jira.percona.com/browse/PMM-7612): Integrated Alerting: Validate 'From' email address format (Settings → Communication → Email) 

Author: ```suggestion
- [PMM-7612](https://jira.percona.com/browse/PMM-7612): Integrated Alerting: Validate communication settings 'From' email address format
```

- [PMM-5291](https://jira.percona.com/browse/PMM-5291): Update ClickHouse version from 19.7.5.27 to 21.3-lts
- [PMM-8091](https://jira.percona.com/browse/PMM-8091): Collect and present additional ProxySQL metrics taken from `runtime_mysql_servers` table

## Bugs Fixed

- [PMM-8458](https://jira.percona.com/browse/PMM-8458): Backup Management: Inconsistent auto-fill of 'Vendor' field with on-demand backup
- [PMM-8404](https://jira.percona.com/browse/PMM-8404): Dashboard image rendering plugin renders image that includes error message
- [PMM-7286](https://jira.percona.com/browse/PMM-7286): Query Analytics can't handle colon character (`:`) in service names (Thanks to Daniel Guzman Burgos for reporting this issue)
- [PMM-7278](https://jira.percona.com/browse/PMM-7278): `mongo_exporter` fails to authorize when MongoDB running with `authMechanism=MONGODB-X509` (Thanks to Lena D for reporting this issue)
- [PMM-8307](https://jira.percona.com/browse/PMM-8307): Default config limits for allowed connections prevents monitoring large numbers (500+) of DB servers
- [PMM-2168](https://jira.percona.com/browse/PMM-2168): `rds_exporter` not stopped when all RDS instances are removed or disabled

> - [PMM-8219](https://jira.percona.com/browse/PMM-8219): PMM Server update panel "Current version" empty if no internet connectivity

Reviewer 2: ```suggestion
- [PMM-8219](https://jira.percona.com/browse/PMM-8219): PMM Server update panel "Current version" fails to display if no internet connectivity
```

Author: The display is `Current version: ()`. There isn't strictly a failure or error message. It just doesn't print anything. I don't want to suggest a cause but merely report the fact of the ticket.

> - [PMM-8581](https://jira.percona.com/browse/PMM-8581): Grafana is not started after PMM upgrade in Docker
> - [PMM-8573](https://jira.percona.com/browse/PMM-8573): Inventory page returns Internal server error after upgrade

Reviewer 3: ```suggestion
```
Adding SKIP.
Both of them defects from 2.21
